src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsListener.java

```diff
@@ -2,65 +2,110 @@
 
 import java.awt.event.KeyAdapter;
 import java.awt.event.KeyEvent;
+import java.util.ArrayList;
+import java.util.Locale;
+import java.util.stream.Collectors;
 
 /**
  * respond to grabKey and display the key binding
  */
 public class KeyBindingsListener extends KeyAdapter {
 
     private final KeyBindingTable table;
+    private boolean isDeleteKey;
+    private boolean isFunctionKey;
+    private boolean isEscapeKey;
 
     public KeyBindingsListener(KeyBindingTable table) {
         this.table = table;
     }
 
     @Override
     public void keyPressed(KeyEvent evt) {
-        // first check if anything is selected if not the return
+        // first check if anything is selected if not then return
         final int selRow = table.getSelectedRow();
         boolean isAnyRowSelected = selRow >= 0;
         if (!isAnyRowSelected) {
             return;
         }
 
-        final String modifier = KeyEvent.getKeyModifiersText(evt.getModifiers());
+        boolean isDeleteKey = false;
+        boolean isFunctionKey = false;
+        boolean isEscapeKey = false;
+        final String modifier = getModifierText(evt);
 
         // VALIDATE code and modifier
         // all key bindings must have a modifier: ctrl alt etc
         if ("".equals(modifier)) {
             int kc = evt.getKeyCode();
-            boolean isFunctionKey = (kc >= KeyEvent.VK_F1) && (kc <= KeyEvent.VK_F12);
-            boolean isEscapeKey = kc == KeyEvent.VK_ESCAPE;
-            boolean isDeleteKey = kc == KeyEvent.VK_DELETE;
+            isFunctionKey = (kc >= KeyEvent.VK_F1) && (kc <= KeyEvent.VK_F12);
```

Comment:
why not keeping the delcaration locally? I don't see any usage outside the if statement.